extensions/prototype-content-with-issues-table/app.R

```diff
@@ -0,0 +1,197 @@
+        mutate(exit_code = as.character(exit_code)) |>
+        mutate(exit_code = case_when(
+               exit_code == "1" ~ "Ended with errors",
+               exit_code == "127" ~ "Command could not be executed outside container",
+               exit_code == "137" ~ "Application terminated due to low memory",
+               exit_code == "2" ~ "Command encountered an error during execution",
+               exit_code == "134" ~ "Terminated due to critical error, potential segfault",
+               exit_code == "255" ~ "Abnormal termination of process",
+               exit_code == "130" ~ "Process terminated by user-initiated signal",
+               exit_code == "13" ~ "File permissions issue",
+               exit_code == "15" ~ "Process terminated externally",
+               TRUE ~ exit_code))
+    },
+    error = function(e) {
+      # content item does not have any jobs 
+      NULL
+    }
+  )
+  
+  if (is.null(failed_jobs) || nrow(failed_jobs) == 0) {
+    return(NULL)
+  } else {
+    print(item)
+    print(failed_jobs)
+    # handle content without usage data, such as unpublished content
+    last_visit <- usage %>%
+      filter(content_guid == item$content$guid) %>%
+      slice_max(timestamp) %>%
+      select(user_guid, timestamp)
+    # some content with issues do not have visit data (ex: unpublished)
+    if (nrow(last_visit) == 0){
+      visit_timestamp <- as.POSIXct(0) 
+      visitor_username <- "No visit data exists for this content"
+    } else {
+      visit_timestamp <- last_visit$timestamp
+      visitor_username <- get_user_info(client, last_visit$user_guid, connectServer)
+    }
+    if (is.na(item$content$title)) {
+      item$content$title <- "" # use empty strings when content is missing title
+    }
+    # return required information from https://github.com/posit-dev/connect/issues/30288 
+    all_failed_jobs <- bind_rows(lapply(seq_len(nrow(failed_jobs)), function(i) {
+      tibble(
+        "content_title" = item$content$title,
+        "content_guid" = item$content$guid,
+        "content_owner" = item$content$owner[[1]]$username,
+        "job_failed_at" = failed_jobs$end_time[i],
+        "failed_job_type" = failed_jobs$tag[i],
+        "failure_reason" = failed_jobs$exit_code[i],
+        "last_deployed_time" = item$content$last_deployed_time,
+        "last_viewed_by" = visitor_username,
+        "last_viewed_time" = visit_timestamp
+      )
+    }))
+    print(all_failed_jobs)
+    all_failed_jobs
+  }
+}
+
+server <- function(input, output, session) {
+  # use API key environment variables for now
+  # we can add local run and connect API integration functionality later
+  api_key <- Sys.getenv("CONNECT_API_KEY")
+  connect_server <- Sys.getenv("CONNECT_SERVER")
+  # initialize Connect API client
+  client <- connect(server = connect_server, api_key = api_key)
+  # set dashboard URL, needed for call to v1/users for visitor user_guid replacement
+  connectServer <- client$get_dashboard_url()
```

Comment:
Doesn't `connectapi` use these by default? Do we need to load them + pass them like this?

Reply:
[Looks like it does!](https://github.com/posit-dev/connectapi/tree/ea0a1401d5ca30a5faabbbc48026a8b832c591ff?tab=readme-ov-file#client) `connectServer` is also cleaned up with the removal of `get_user_info()` as described in [another comment](https://github.com/posit-dev/connect-extensions/pull/33#pullrequestreview-2672118035), so this entire block will be tidied.